source/extensions/filters/http/cache/http_cache.h

```diff
@@ -175,7 +175,7 @@ size_t stableHashKey(const Key& key);
 // to determine what ranges are needed.
 class LookupRequest {
 public:
-  using HeaderVector = std::vector<Http::HeaderEntry>;
+  using HeaderVector = std::vector<const Http::HeaderEntry*>;
```

Comment:
I was planning to move from `vector<const HeaderEntry*>` to `flat_hash_map<string,string>` because I couldn't find a straightforward way to get all entries for a header with multiple instances.
To handle multiple instances of the same header in the headermap, I would use `Http::HeaderUtility::getAllOfHeader` which adds all values to a `vector<string_view>`, combine the values and add a entry in the map `header_name -> combined_value`.

The main problems are that (i) `HeaderEntry` is an abstract class, so we can't create a new instance to combine the values, we can only get the pointer using `get`; and (ii) `get` only returns the first instance, so I couldn't find a way to get the multiple instances in `HeaderEntry` form.

The only alternative I could find to maybe stick with `vector<HeaderEntry*>` was to create a copy of `getAllOfHeader` to populate the vector directly, but even that didn't work straightaway.
I could give it another try if you rather have the vector than the map.

Reply:
If we're going to give plugins access to the raw headers, we may as well just replace this with a const RequestHeaderMap&, with warnings that it mustn't be accessed off-thread.